src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverFactoryActivator.java

```diff
@@ -79,6 +81,45 @@ private static final class FactoryRegistration {
         public volatile CommonResourceResolverFactoryImpl commonFactory;
     }
 
+    private static final class VanityPathConfigurer {
+        private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+        void configureVanityPathPrefixes(String[] pathPrefixes, String[] pathPrefixesFallback,
```

Comment:
I wonder if it makes sense to pass in the whole config object, so simplify the signature ; now we pass to fields and their names, passing in the object might make things nicer.

Reply:
I attempted to implement this suggestion and the result looks like below. I don't think it's all that nicer,what do you think?
`void configureVanityPAthPrefix(ResourceResolverFactoryConfig config, Consumer<String[]> allowVanityPathConsumer,  
                                   Consumer<String[]> denyVanityPathConsumer) {
        configureVanityPathPrefixes(config.resource_resolver_vanitypath_whitelist(),
            config.resource_resolver_vanitypath_allowlist(),
            "resource_resolver_vanitypath_whitelist",
            "resource_resolver_vanitypath_allowlist",
            allowVanityPathConsumer);
        configureVanityPathPrefixes(config.resource_resolver_vanitypath_blacklist(),
            config.resource_resolver_vanitypath_denylist(),
            "resource_resolver_vanitypath_blacklist",
            "resource_resolver_vanitypath_denylist",
            denyVanityPathConsumer);
    }`